testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -195,10 +197,12 @@ public void testViewColumnAccessControl()
                 privilege(viewOwnerSession.getUser(), "orders", CREATE_VIEW_WITH_SELECT_COLUMNS));
 
         // verify the view owner can select from the view even without special view creation privileges
+/* TODO: this is no longer true! the owner has no special privileges anymore
```

Comment:
How that happened?

Reply:
Previously, if the current session's user was the view's owner, it fell into this branch in `StatementAnalyzer`:

```java
                else {
                    identity = session.getIdentity();
                    viewAccessControl = accessControl;
                }
```

Which means that the view owner was treated as if the view was `SECURITY INVOKER`. Which goes slightly further than "the current session's enabled roles are used".

Honestly, I wasn't expecting this. Now the question is if this is the expected semantics which should be preserved (entirely or in part).

Reply:
It is expected semantics. View owner should have the access to its view.

Reply:
What about https://gist.github.com/kokosing/3b0dabfda2d85f08c03c7d669dd06c50? I am not sure if we should repeat that for connector access controls.

Reply:
I am not sure about `.withEnabledRoles(metadata.listRoles(session, Optional.empty()))`, but otherwise seems like the way to go

Reply:
> I am not sure about...

By that I mean hat it should be delegated to the system access control instead of assuming that surely we want all granted roles to be enabled.